test/ResultSet/ResultSetIntegrationTest.php

```diff
@@ -90,6 +90,20 @@ public function testCanProvideIteratorAsDataSource()
         $this->assertSame($it, $this->resultSet->getDataSource());
     }
 
+    public function testCanProvideArrayAsDataSource()
+    {
+        $dataSource = [['foo']];
+        $this->resultSet->initialize($dataSource);
+        $this->assertEquals($dataSource[0], (array) $this->resultSet->current());
+
+        $returnType = new ReturnType();
+        $dataSource = [$returnType];
+        $this->resultSet->setArrayObjectPrototype($returnType);
```

Comment:
@bartmcleod I created PR to your PR branch https://github.com/bartmcleod/zend-db/pull/1